spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/dynamic/SleuthDynamicAspectJTracingInterceptor.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.sleuth.dynamic;
+
+import brave.ScopedSpan;
+import brave.Tracer;
+import org.aopalliance.intercept.MethodInvocation;
+
+import org.springframework.aop.IntroductionInterceptor;
+import org.springframework.aop.support.AopUtils;
+import org.springframework.cloud.sleuth.util.SpanNameUtil;
+
+/**
+ * Interceptor for tracing {@link MethodInvocation} objects.
+ *
+ * @author Taras Danylchuk
+ * @since 2.2.0
+ */
+public class SleuthDynamicAspectJTracingInterceptor implements IntroductionInterceptor {
+
+	private static final String CLASS_NAME = "Class name";
```

Comment:
as tags are lookup keys and subject to indexing concerns, I think these sorts of names won't be good choices.

Reply:
Class-name, method-name ?

Reply:
I've reused existing logic from annotations feature
although it has only simple name instead of full one for class name
```invocation.getThis().getClass().getSimpleName()``` @marcingrzejszczak any reason for that?